### PR TITLE
RR-2785 - Set PES as the CIAG KPI rule for integration tests

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
Set PES as the CIAG KPI rule for integration tests. Set up separate integration tests for PEF

### CODE REVIEW

File: src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReceivedDueToAdmissionEventTest.kt [hunk 1]
```diff
@@ -116,6 +120,111 @@ class PrisonerReceivedDueToAdmissionEventTest : IntegrationTestBase() {
...
+    with(aValidPrisoner(prisonerNumber = prisonNumber, prisonId = "MDI")) {
+      createPrisonerAPIStub(prisonNumber, this)
+    }
+
+    assertThat(runCatching { getInduction(prisonNumber) }.getOrNull()).isNull()
+    assertThat(runCatching { getInductionSchedule(prisonNumber) }.getOrNull()).isNull()
+
+    val earliestDateTime = OffsetDateTime.now()
+
+    educationAssessmentEventRepository.save(
```
Comment (Mary-ford) [inline src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReceivedDueToAdmissionEventTest.kt:136]: I think this implies that the prisoner has had their S&A completed, and then we get an event saying they have been admitted to prison.  I would have expected the S&A would happen after we get the admission event? How does this scenario happen?

File: src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/specialtests/PrisonerReceivedDueToAdmissionEventDuringChristmasTest.kt [hunk 2]
```diff
@@ -82,13 +89,119 @@ class PrisonerReceivedDueToAdmissionEventDuringChristmasTest : IntegrationTestBa
...
+        }
+
+      val inductionScheduleEvents = inductionScheduleEventQueue.receiveEventsOnQueue(QueueType.INDUCTION)
+      assertThat(inductionScheduleEvents)
+        .hasNumberOfEvents(2)
+        .allEvents {
+          it.hasDetailUrl("http://localhost:8080/inductions/$prisonNumber/induction-schedule")
+            .hasNumberOfPersonReferenceIdentifiers(1)
+            .personReferenceIdentifier(1) { it.hasValue(prisonNumber) }
+        }
```
Comment (Mary-ford) [inline src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/specialtests/PrisonerReceivedDueToAdmissionEventDuringChristmasTest.kt:204]: Should this send an outbound event to notify that the induction schedule has been created?

File: src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReceivedDueToAdmissionEventTest.kt (hunk 1 above)
Comment (nathanrussell-moj-digital) [reply src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReceivedDueToAdmissionEventTest.kt:136]: Brilliant question, thats exactly what it means, and yes your expectation of "normal" events is correct 👍 

> I think this implies that the prisoner has had their S&A completed, and then we get an event saying they have been admitted to prison

...
> How does this scenario happen?

It's basically an edge case, but technically is possible.
All of our processing in this area is message driven. We receive and process events from NOMIS/Domain Events telling us stuff has happened about the prisoner (such as admissions), and we also get messages from Curious saying that the S&As are complete.

Messaging is asynchronous. There is no guarantee when we'll get a message, or in what order messages will arrive (but common sense / logic states that we _should_ get the admission message from NOMIS before the S&A message from Curious)
But sometimes things don't always go to plan:
* Perhaps NOMIS/Domain Events didnt send the admission message at all? (rare, and would be an exceptional case, also also MN listen to the same event so in theory would not have setup the prisoner record at their end)
* Perhaps LWP failed to process the admission event for some reason. Maybe we retried it 3 times, and then it landed on the DLQ
* Maybe LWP entirely was down for a weekend and nobody knew until Monday. During that time NOMIS/Domain Events sent the admission message, and then at some point later Curious sent the S&A message. Both messages are sat on the LWP queues (a different queue for each), but not processed because LWP is down. Then on Monday when we restart LWP, the messages get processed ... but the order of receipt and processing by LWP is not guaranteed.
* and probably some other weird and whacky scenarios ...

It's technically possible, but is very much an edge case. Having a test around it at least shows we've thought about it and have a predictable outcome in this scenario.

> I would have expected the S&A would happen after we get the admission event?

100% yes, that is the "normal" sequence of events that we would expect 👍

File: src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/specialtests/PrisonerReceivedDueToAdmissionEventDuringChristmasTest.kt (hunk 2 above)
Comment (nathanrussell-moj-digital) [reply src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/specialtests/PrisonerReceivedDueToAdmissionEventDuringChristmasTest.kt:204]: Yes, thats what one of the 2 messages are. The first is sent when the `InductionSchedule` was created with a status of `PENDING_INITIAL_SCREENING_AND_ASSESSMENTS_FROM_CURIOUS` and the 2nd was sent when the `InductionSchedule` was updated with a status of `SCHEDULED`

The assertions just above this on the `InductionScheduleHistory` assert that there are 2 `InductionSchedules` and that they have the correct statuses/deadline etc.

The assertions here to do with the message sent to MN is the right / appropriate way to do it.
The semantics of the message sent to MN is that every time an `InductionSchedule` is changed in any way (inc. initial creation), we send a message to MN saying "the prisoner's InductionSchedule has changed; come and get the history records if you want the detail"
The message itself does not contain anything about the InductionSchedule, so we cant write assertions here to that effect.
All we can really do is assert that the correct number of messages was sent for the prisoner in question.

Comment (Mary-ford) [reply src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/specialtests/PrisonerReceivedDueToAdmissionEventDuringChristmasTest.kt:204]: Oh yes, sorry.  I'd missed that this was doing that check. Makes sense, thanks.

**Review by Mary-ford — APPROVED** [review]
LGTM
